testutils/resourcebuilder/machine/v1beta1/aws_provider_spec.go

```diff
@@ -127,26 +154,76 @@ func (m AWSProviderSpecBuilder) BuildRawExtension() *runtime.RawExtension {
+func (m AWSProviderSpecBuilder) WithAMI(ami machinev1beta1.AWSResourceReference) AWSProviderSpecBuilder {
+	m.ami = &ami
+	return m
+}
+
 // WithAvailabilityZone sets the availabilityZone for the AWS machine config builder.
 func (m AWSProviderSpecBuilder) WithAvailabilityZone(az string) AWSProviderSpecBuilder {
-	m.availabilityZone = az
+	m.availabilityZone = &az
+	return m
+}
+
+// WithBlockDevices sets the BlockDevices for the AWS machine config builder.
+// Use WithBlockDevices(nil), to set them to the zero value.
+func (m AWSProviderSpecBuilder) WithBlockDevices(blockDevices []machinev1beta1.BlockDeviceMappingSpec) AWSProviderSpecBuilder {
+	m.blockDevices = &blockDevices
+	return m
+}
+
+// WithCredentialsSecret sets the CredentialsSecret for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithCredentialsSecret(credentialsSecret *corev1.LocalObjectReference) AWSProviderSpecBuilder {
+	m.credentialsSecret = &credentialsSecret
+	return m
+}
+
+// WithIAMInstanceProfile sets the AMI for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithIAMInstanceProfile(iamInstanceProfile *machinev1beta1.AWSResourceReference) AWSProviderSpecBuilder {
+	m.iamInstanceProfile = &iamInstanceProfile
 	return m
 }
 
 // WithInstanceType sets the instanceType for the AWS machine config builder.
 func (m AWSProviderSpecBuilder) WithInstanceType(instanceType string) AWSProviderSpecBuilder {
-	m.instanceType = instanceType
+	m.instanceType = &instanceType
+	return m
+}
+
+// WithLoadBalancers sets the LoadBalancers for the AWS machine config builder.
+// Use WithLoadBalancers(nil), to set them to the zero value.
+func (m AWSProviderSpecBuilder) WithLoadBalancers(loadBalancers []machinev1beta1.LoadBalancerReference) AWSProviderSpecBuilder {
+	m.loadBalancers = &loadBalancers
+	return m
+}
+
+// WithPlacement sets the Placement for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithPlacement(placement machinev1beta1.Placement) AWSProviderSpecBuilder {
+	m.placement = &placement
+	return m
+}
+
+// WithRegion sets the region for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithRegion(region string) AWSProviderSpecBuilder {
+	m.region = &region
 	return m
 }
 
 // WithSecurityGroups sets the securityGroups for the AWS machine config builder.
 func (m AWSProviderSpecBuilder) WithSecurityGroups(sgs []machinev1beta1.AWSResourceReference) AWSProviderSpecBuilder {
-	m.securityGroups = sgs
+	m.securityGroups = &sgs
 	return m
 }
 
 // WithSubnet sets the subnet for the AWS machine config builder.
 func (m AWSProviderSpecBuilder) WithSubnet(subnet machinev1beta1.AWSResourceReference) AWSProviderSpecBuilder {
-	m.subnet = subnet
+	m.subnet = &subnet
+	return m
+}
+
+// WithUserDataSecret sets the UserDataSecret for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithUserDataSecret(userDataSecret *corev1.LocalObjectReference) AWSProviderSpecBuilder {
```

Comment:
Can you add a test that shows what happens when I pass `WithUserDataSecret(nil)`?